### PR TITLE
[Stopwatch] add Helper and getFormattedDuration methods

### PR DESCRIPTION
**Pull request by mixo**
Q | A
-- | --
Branch? | master
Bug fix? | no
New feature? | yes
BC breaks? | no
Deprecations? | no
Tests pass? | yes
Fixed tickets | 
License | MIT
Doc PR |

### CODE REVIEW

File: src/Symfony/Component/Stopwatch/Helper.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Stopwatch;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
```
Comment (OskarStark) [inline src/Symfony/Component/Stopwatch/Helper.php]: Is this true?

File: src/Symfony/Component/Stopwatch/Helper.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ */
+class Helper
+{
+    /**
+     * @param int $time
```
Comment (OskarStark) [inline src/Symfony/Component/Stopwatch/Helper.php]: This can be removed

File: src/Symfony/Component/Stopwatch/Helper.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @return string
```
Comment (OskarStark) [inline src/Symfony/Component/Stopwatch/Helper.php]: Can we use a return type instead?

File: src/Symfony/Component/Stopwatch/StopwatchEvent.php [hunk 4]
```diff
@@ -193,6 +193,16 @@ public function getDuration()
         return $total;
     }
 
+    /**
+     * Gets the formatted duration of the events (including all periods).
+     *
+     * @return string The formatted duration
+     */
+    public function getFormattedDuration()
```
Comment (OskarStark) [inline src/Symfony/Component/Stopwatch/StopwatchEvent.php]: Can we use a return type here ?

File: src/Symfony/Component/Stopwatch/StopwatchPeriod.php [hunk 5]
```diff
@@ -64,6 +64,16 @@ public function getDuration()
         return $this->end - $this->start;
     }
 
+    /**
+     * Gets the formatted time spent in this period.
+     *
+     * @return string The formatted period duration
+     */
+    public function getFormattedDuration()
```
Comment (OskarStark) [inline src/Symfony/Component/Stopwatch/StopwatchPeriod.php]: Can we use a return type here ?

File: src/Symfony/Component/Stopwatch/Tests/HelperTest.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Stopwatch\Helper;
+
+class HelperTest extends TestCase
+{
+    /**
+     * @dataProvider provideFormattedTime
+     */
+    public function testFormatTime($time, $formattedTime)
```
Comment (OskarStark) [inline src/Symfony/Component/Stopwatch/Tests/HelperTest.php]: Please use “ int $time, string $formattedTime”

**Comment by ro0NL** [comment]
i tend to believe formatting is more of a templating concern, and Symfony should opt-out from any low-level format IMHO

nevertheless it could bet worth exploring if the profiler time panel template can use this? Doing this from e.g. its data provider is fine.

**Review by OskarStark — COMMENTED** [review]
Just a few comments

**Comment by alexey-okhorzin** [comment]
The tests failed but this isn't related to my changes

**Review by OskarStark — APPROVED** [review]

**Comment by fabpot** [comment]
I agree that formatting is out of the scope of the component. Thanks for proposing.

**Pull request closed without merging**